homeassistant/components/zwave_js/light.py

```diff
@@ -426,3 +433,73 @@ def _calculate_color_values(self) -> None:
             self._rgbw_color = (red, green, blue, white)
             # Light supports rgbw, set color mode to rgbw
             self._color_mode = COLOR_MODE_RGBW
+
+
+class ZwaveBlackIsOffLight(ZwaveLight):
+    """
+    Representation of a Z-Wave light where setting the color to black turns it off.
+
+    Currently only supports lights with RGB, no color temperature, and no white channels.
+    """
+
+    def __init__(
+        self, config_entry: ConfigEntry, client: ZwaveClient, info: ZwaveDiscoveryInfo
+    ) -> None:
+        """Initialize the light."""
+        super().__init__(config_entry, client, info)
+
+        self._last_color: dict[str, int] | None = None
+        self._supported_color_modes.discard(COLOR_MODE_BRIGHTNESS)
+
+    @property
+    def brightness(self) -> int:
+        """Return the brightness of this light between 0..255."""
+        return 255
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if device is on (brightness above 0)."""
+        return any(value != 0 for value in self.info.primary_value.value.values())
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the device on."""
+        await super().async_turn_on(**kwargs)
+
+        if (
+            kwargs.get(ATTR_RGBW_COLOR) is None
+            and kwargs.get(ATTR_COLOR_TEMP) is None
+            and kwargs.get(ATTR_HS_COLOR) is None
```

Comment:
Side note: We should consider changing our supported color mode to `COLOR_MODE_RGB`. Nowadays the light integration will set `ATTR_RGB_COLOR` in the service handler parameters if any of the other color parameters are sent in the service call and the light has color mode `COLOR_MODE_RGB` but not the color mode of the passed color parameter.

Reply:
I'd invert this check and return if true. Then we can outdent below.